File: tmp/debug_db.py
import os
import psycopg2
from dotenv import load_dotenv

load_dotenv()

conn_str = os.getenv("NEXT_PUBLIC_NEON_DB_CONNECTION_STRING")
if not conn_str:
    print("Error: NEXT_PUBLIC_NEON_DB_CONNECTION_STRING not found in .env")
    exit(1)

try:
    conn = psycopg2.connect(conn_str)
    cur = conn.cursor()
    cur.execute("SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'")
    tables = cur.fetchall()
    print("Tables in database:", [t[0] for t in tables])
    
    cur.execute("SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'roadmaps'")
    columns = cur.fetchall()
    print("Columns in 'roadmaps':", columns)
    
except Exception as e:
    print("Error connecting to database or querying:", e)
finally:
    if 'conn' in locals():
        conn.close()

File: tmp/debug_db.ts
import { db } from "../lib/db/db";
import { roadmapsTable } from "../lib/db/schema";
import { sql } from "drizzle-orm";

async function debug() {
    try {
        console.log("Checking database connection...");
        const result = await db.execute(sql`SELECT table_name FROM information_schema.tables WHERE table_schema = 'public'`);
        console.log("Tables found:", result.rows.map(r => r.table_name));

        const columns = await db.execute(sql`SELECT column_name, data_type FROM information_schema.columns WHERE table_name = 'roadmaps'`);
        console.log("Columns in 'roadmaps':", columns.rows);

        process.exit(0);
    } catch (err) {
        console.error("Database Debug Error:", err);
        process.exit(1);
    }
}

debug();
